src/hotspot/share/runtime/fieldDescriptor.cpp

```diff
@@ -216,18 +207,33 @@ void fieldDescriptor::print_on_for(outputStream* st, oop obj) {
       ShouldNotReachHere();
       break;
   }
-  // Print a hint as to the underlying integer representation. This can be wrong for
-  // pointers on an LP64 machine
+
+  // Print  a hint as to the underlying integer representation.
+  if (is_reference_type(ft)) {
 #ifdef _LP64
-  if (is_reference_type(ft) && UseCompressedOops) {
-    st->print(" (%x)", obj->int_field(offset()));
-  }
-  else // <- intended
+    if (UseCompressedOops) {
+      st->print(" (" PTR32_FORMAT ")", obj->int_field(offset()));
+    } else {
+      st->print(" (" PTR_FORMAT ")", obj->long_field(offset()));
+    }
+#else
+    st->print(" (" PTR32_FORMAT ")", obj->int_field(offset()));
 #endif
-  if (ft == T_LONG || ft == T_DOUBLE LP64_ONLY(|| !is_java_primitive(ft)) ) {
-    st->print(" (%x %x)", obj->int_field(offset()), obj->int_field(offset()+sizeof(jint)));
-  } else if (as_int < 0 || as_int > 9) {
-    st->print(" (%x)", as_int);
+  } else { // Primitives
+    if (ft == T_LONG || ft == T_DOUBLE) {
+      st->print(" (" PTR64_FORMAT ")", obj->long_field(offset()));
+    } else {
+      switch (ft) {
+        case T_BYTE:    st->print(" (" PTR8_FORMAT  ")", obj->byte_field(offset()));  break;
+        case T_CHAR:    st->print(" (" PTR16_FORMAT ")", obj->char_field(offset()));  break;
+        case T_FLOAT:   st->print(" (" PTR32_FORMAT ")", obj->int_field(offset()));   break;
+        case T_INT:     st->print(" (" PTR32_FORMAT ")", obj->int_field(offset()));   break;
```

Comment:
Should these be collapsed into one case, as implicitly done for long/double?

Reply:
I see your point. I think I prefer to change the long/double cases to look like the rest of the block of primitives. I think this gets easier to read if we remove that if/else. Take a look and see if you agree.